modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/declaration/DefaultXmlArtifactDeclarationLoader.java

```diff
@@ -450,11 +453,21 @@ private void declareTransform(ComponentModel model,
 
     // add DW script text content to script parameter
     if (groupConfig.getTextContent() != null) {
-      objectBuilder.withParameter(SCRIPT, ParameterSimpleValue.of(groupConfig.getTextContent()));
+      ParameterValue value;
+      if (isCData(groupConfig)) {
+        value = plain(groupConfig.getTextContent());
+      } else {
+        value = cdata(groupConfig.getTextContent());
+      }
+      objectBuilder.withParameter(SCRIPT, value);
     }
     return objectBuilder;
   }
 
+  private boolean isCData(ConfigLine config) {
+    return config.getCustomAttributes().get(IS_CDATA) != null;
```

Comment:
sounds like get(String) should return Optional. Evaluate if this will become a backwards concern and is something that can be changed later